File: receiver.js
var zmq = require('zmq')
var port = 3000;

var socket = zmq.socket('rep');

  socket.identity = 'server' + process.pid;

  socket.bind("tcp://127.0.0.1:"+port, function(err) {
    if (err) throw err;
    console.log("Bound to port "+port+".");

    socket.on('message', function(data) {
      //console.log(socket.identity + ': received ' + data.toString());
      console.log("Received: "+data.toString());
      data = JSON.parse( data );
      var A = data["A"];
      var B = data["B"];
      //console.log(A, B);
      var answer = Math.floor( A / B );
      console.log("Sent: "+answer);
      socket.send(answer);
    });
});

File: receiver.py
import zmq
import time
import sys

port = "3000"
if len(sys.argv) > 1:
    port =  sys.argv[1]
    int(port)


context = zmq.Context()
socket = context.socket(zmq.REP)

socket.bind("tcp://127.0.0.1:%s" % port)
print "Bound to port %s." % port

while True:
    message = socket.recv()
    socket.send(message + " Blancmange!")
